rcl/src/rcl/security_directory.c

```diff
@@ -112,12 +104,16 @@ static bool get_best_matching_directory(
         memcpy(matched_name, file.name, max_match_length);
       }
     }
-    if (-1 == tinydir_next(&dir)) {
+    rcutils_file_fini(&file, dir.allocator);
+    if (RCUTILS_RET_OK != rcutils_next_dir(&dir)) {
       goto cleanup;
     }
   }
+
 cleanup:
-  tinydir_close(&dir);
+  if (rcutils_close_dir(&dir) != RCUTILS_RET_OK) {
+    RCL_SET_ERROR_MSG("Error closing the directory");
```

Comment:
It would be helpful to include the directory path in the error message.

Reply:
Included directory path in the error message: https://github.com/ros2/rcl/pull/605/commits/de50b686dc6f8d5fa2963ddee842bdcc89827bca